recipes/botan/all/conanfile.py

```diff
@@ -0,0 +1,320 @@
+                       'system_cert_bundle': None}
+
+    def configure(self):
+        msvc_too_old = self.settings.os == "Windows" and \
+                       self.settings.compiler == "Visual Studio" and \
+                       Version(self.settings.compiler.version.value) < "14"
+        if msvc_too_old:
+            raise ConanInvalidConfiguration("Botan doesn't support MSVC < 14")
+
+        if self._needs_cxx_abi_workaround:
+            self.output.warn("Overriding compiler.libcxx setting to use libstdc++11")
+            self.settings.compiler.libcxx = 'libstdc++11'
+
+        if self.options.boost:
+            self.options["boost"].add("shared=False")
+            self.options["boost"].add("magic_autolink=False")
+            self.options["boost"].add("without_coroutine=False")
+            self.options["boost"].add("without_system=False")
+
+    def build_requirements(self):
+        if self.settings.os == "Windows":
+            self.build_requires("jom_installer/1.1.2@bincrafters/stable")
+
+    def requirements(self):
+        if self.options.bzip2:
+            self.requires("bzip2/1.0.6")
+        if self.options.openssl:
+            self.requires("OpenSSL/1.0.2s")
+        if self.options.zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.sqlite3:
+            self.requires("sqlite3/3.30.1")
+        if self.options.boost:
+            self.requires("boost/1.70.0")
+
+    def config_options(self):
+        if self.options.single_amalgamation:
+            self.options.amalgamation = True
+
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "botan-" + self.version
+        os.rename(extracted_dir, "sources")
+
+    def build(self):
+        with tools.chdir('sources'):
+            self.run(self._configure_cmd)
+            self.run(self._make_cmd)
+
+    def package(self):
+        self.copy(pattern="license.txt", dst="licenses", src="sources")
+        with tools.chdir("sources"):
+            self.run(self._make_install_cmd)
+
+    def package_info(self):
+        if self.settings.compiler == 'Visual Studio':
+            self.cpp_info.libs.append('botan')
+        else:
+            self.cpp_info.libs.extend(['botan-2'])
+            if self.settings.os != 'Windows':
+                self.cpp_info.libs.append('dl')
+            if self.settings.os == 'Linux':
+                self.cpp_info.libs.append('rt')
+            if self.settings.os == 'Macos':
+                self.cpp_info.exelinkflags = ['-framework Security', '-framework CoreFoundation']
+            if not self.options.shared:
+                self.cpp_info.libs.append('pthread')
+        if self.settings.os == "Windows":
+            self.cpp_info.libs.extend(["ws2_32", "Crypt32"])
+
+        self.cpp_info.libdirs = ['lib']
+        self.cpp_info.bindirs = ['lib', 'bin']
```

Comment:
why lib is bindir?

Reply:
This is for Windows. Otherwise `botan.dll` is not found in PATH.

Reply:
windows dlls should be placed in the bin directory

Reply:
Just to be clear, you're saying it's fine as it is, right?

Reply:
The problem here is that the `lib` folder should not be part of `cpp_info.bindirs` as no binaries should be placed there, only `bin` folder should contain binaries (executables and dlls)

Reply:
Thanks for the clarification, got it now. I'm still trying to figure out how to tell Botan's build system to do so. There's a `--libdir=bin` as an option, but it will place both `botan.dll` and `botan.lib` in the bin folder, which results in the test-package not finding `botan.lib` anymore.

Reply:
I've added a patch for Botan to place the DLL in the binary directory rather
than the library dir. That should work and we'll try to get it upstream in the
next Botan version.